### DISCUSSION TITLE
Support UTF-8 for name of files

### DISCUSSION THREAD

**Discussion by MohammadPCh** [vercel/next.js#14961] [Ideas]
Hi there,
Consider this route:
https://example.com/movie/[movieName]
my site's language is Persian so for example this is a normal URL:
https://example.com/movie/این-یک-اسم-تستی-است-که-طول-آن-خیلی-خیلی-بلند-است-و-مشکل-ساز-میشود-باید-فکری-کرد
when i build the project it throw this error:
Error: ENAMETOOLONG: name too long, open '/home/lobster/projects/website-next/.next/export/movie/33193/%D8%A7%DB%8C%D9%86-%DB%8C%DA%A9-%D8%A7%D8%B3%D9%85-%D8%AA%D8%B3%D8%AA%DB%8C-%D8%A7%D8%B3%D8%AA-%DA%A9%D9%87-%D8%B7%D9%88%D9%84-%D8%A2%D9%86-%D8%AE%DB%8C%D9%84%DB%8C-%D8%AE%DB%8C%D9%84%DB%8C-%D8%A8%D9%84%D9%86%D8%AF-%D8%A7%D8%B3%D8%AA-%D9%88-%D9%85%D8%B4%DA%A9%D9%84-%D8%B3%D8%A7%D8%B2-%D9%85%DB%8C%D8%B4%D9%88%D8%AF-%D8%A8%D8%A7%DB%8C%D8%AF-%D9%81%DA%A9%D8%B1%DB%8C-%DA%A9%D8%B1%D8%AF.json'
This happens because Nextjs tries to save encoded string. my idea is save the original string. because encoded string has 384 characters but original string has only 79 characters and linux has 255 char limit for name of files.
thanks.

**Comment by lonyele** [comment]
Hi! @MohammadPCh  I have a same problem and made an issue(#15858). Did you solve the problem?

**Reply by MohammadPCh** [reply]
Hi @lonyele, unfortunately no! we can add @Timer here. maybe he has an idea.
